tests/suites/test_suite_mpi.function

```diff
@@ -763,6 +763,316 @@ exit:
 }
 /* END_CASE */
 
+/* BEGIN_CASE */
+void mpi_core_cond_assign( data_t * input_X,
+                           data_t * input_Y,
+                           int input_len )
+{
+    mbedtls_mpi_uint *X = NULL;
+    mbedtls_mpi_uint *Y = NULL;
+    size_t limbs_X = CHARS_TO_LIMBS( input_X->len );
+    size_t limbs_Y = CHARS_TO_LIMBS( input_Y->len );
+    size_t limbs = limbs_X;
+    size_t copy_limbs = CHARS_TO_LIMBS( input_len );
+    size_t len = limbs * sizeof( mbedtls_mpi_uint );
+    size_t copy_len = copy_limbs * sizeof( mbedtls_mpi_uint );
```

Comment:
Do I understand correctly that the idea behind having `input_len` is to make sure there is no writing beyond the bounds of the destination buffer?

If so, it seems there is a missing test: for buffer under-write.

(I have seen `memcpy()` implementations that iterate backwards, for which writing beyond the beginning of the buffer would be a reasonable thing to worry about)

Reply:
We don't need any test-specific effort to detect buffer underwrites or overwrites, as long as the allocated buffers have exactly the desired size (as opposed to passing a part of a larger allocation). Asan and Valgrind both detect buffer overflows, on either side, both for stack and heap buffers. We test Asan builds in most of the configurations we test.

Reply:
This benefit from Asan is why we allocate exact sizes for buffers as much as possible. Older code didn't take advantage of this and often used larger-than-necessary stack buffers for convenience, but we shouldn't do that anymore.

Yes, we should write this down somewhere discoverable.

Reply:
I'm going to write something quick for the knowledge base.

Reply:
https://github.com/Mbed-TLS/mbedtls-docs/pull/56

Reply:
These test cases are not really about over and underwriting the memory, it is done by the memory sanitizer. They are about sanity testing the `limbs` parameter. The caller should determine the size of the MPI and these tests are checking the length is not altered by the tested function. I want to write white box tests therefore all parameters should be tested.

You are right about that only the smaller `limbs` value is tested and the bigger value is also need to be checked.

Reply:
Testing a bigger value for  the `limbs` parameter implies a fundamental change for the test code. It needs to simulate a memory overwriting which, I think, is too much for these test cases so I decided to not implement them.